Geometry/EcalAlgo/src/EcalEndcapGeometry.cc

```diff
@@ -516,10 +516,20 @@ EcalEndcapGeometry::avgAbsZFrontFaceCenter() const
    return m_avgZ;
 }
 
-const CaloCellGeometry* 
-EcalEndcapGeometry::cellGeomPtr( uint32_t index ) const
-{
-   const CaloCellGeometry* cell ( &m_cellVec[ index ] ) ;
-   return ( m_cellVec.size() < index ||
-	    nullptr == cell->param() ? nullptr : cell ) ;
+const CaloCellGeometry*
```

Comment:
Having the `cellGeomPtr(index)` method return a `const CaloCellGeometry*` or an `std::shared_ptr<CaloCellGeometry>` depending if the call is `const` or not is probably a bad idea.

Reply:
Now only shared_ptr